src/winforms/toga_winforms/widgets/optioncontainer.py

```diff
@@ -24,7 +24,7 @@ def add_content(self, label, widget):
         widget.AutoSize = True
 
         item.Controls.Add(widget.native)
-        self.native.TabPages.Add(item)
+        self.native.TabPages.Insert(index, item)
```

Comment:
Turns out Winforms doesn't like inserting at a position larger than the current size of the list - including inserting at position 0 in an empty list.